### PR TITLE
dialects: (llvm) Add support for overflow flags

### PR DESCRIPTION
**Pull request by lfrenot**

### CODE REVIEW

File: tests/filecheck/dialects/llvm/arithmetic.mlir [hunk 1]
```diff
@@ -1,47 +1,72 @@
...
 
-    %sdiv = llvm.sdiv %arg0, %arg1 : i32
-    // CHECK: %sdiv = llvm.sdiv %arg0, %arg1 : i32
+%add_both = llvm.add %arg0, %arg1 {"overflowFlags" = #llvm.overflow<nsw, nuw>} : i32
+// CHECK: %add_both = llvm.add %arg0, %arg1 {"overflowFlags" = #llvm.overflow<nsw,nuw>} : i32
 
-    %urem = llvm.urem %arg0, %arg1 : i32
-    // CHECK: %urem = llvm.urem %arg0, %arg1 : i32
+%add_both_reverse = llvm.add %arg0, %arg1 {"overflowFlags" = #llvm.overflow<nuw, nsw>} : i32
+// CHECK: %add_both_reverse = llvm.add %arg0, %arg1 {"overflowFlags" = #llvm.overflow<nsw,nuw>} : i32
```
Comment (superlopuh) [inline tests/filecheck/dialects/llvm/arithmetic.mlir:24]: I'm confused, aren't these supposed to be printed with the custom syntax?

Comment (lfrenot) [reply tests/filecheck/dialects/llvm/arithmetic.mlir:24]: Oh, you're probably correct
I'll look into it

File: xdsl/dialects/llvm.py [hunk 2]
```diff
@@ -396,18 +396,96 @@ def print(self, printer: Printer) -> None:
...
+    def parse(cls, parser: Parser):
+        lhs = parser.parse_unresolved_operand()
+        parser.parse_characters(",")
+        rhs = parser.parse_unresolved_operand()
+        overflowFlags = cls.parse_overflow(parser)
+        attributes = parser.parse_optional_attr_dict()
+        if "overflowFlags" in attributes:
+            flags = attributes.pop("overflowFlags")
+            if isinstance(flags, OverflowAttr):
+                overflowFlags = flags
```
Comment (superlopuh) [inline xdsl/dialects/llvm.py:463]: What's MLIR's behaviour here?

Comment (lfrenot) [reply xdsl/dialects/llvm.py:461]: I don't know
And I'm not sure how to check for that

**Comment by lfrenot** [comment]
@tobiasgrosser Are we good to merge?

**Comment by tobiasgrosser** [comment]
> @tobiasgrosser Are we good to merge?

Yes

**Comment by superlopuh** [comment]
I'm afraid it wasn't ready, the IR doesn't roundtrip, and there are still some questions about printing and parsing the overflow flags when they are specified both in the custom syntax and in the attribute dictionary. I would have appreciated a ping.

**Comment by tobiasgrosser** [comment]
Oh sorry, I confused this with the lean-mlir repo.

**Pull request merged**